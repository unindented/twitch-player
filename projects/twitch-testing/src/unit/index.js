import { ThemeProvider } from "@twitch-player/themes/dist/context";
import { light as lightTheme } from "@twitch-player/themes/dist/themes";
import React from "react";
import { render } from "react-testing-library";

const customRender = (node, ...options) => {
  return render(
    <ThemeProvider theme={lightTheme}>{node}</ThemeProvider>,
    ...options
  );
};

const fireResizeEvent = (el, { left = 0, top = 0, width = 0, height = 0 }) => {
  Object.defineProperties(window.HTMLElement.prototype, {
    offsetLeft: { get: () => left },
    offsetTop: { get: () => top },
    offsetWidth: { get: () => width },
    offsetHeight: { get: () => height },
  });

  const evt = document.createEvent("Event");
  evt.initEvent("resize", true, true);
  el.dispatchEvent(evt);
};

export * from "react-testing-library";

export { customRender as render, fireResizeEvent };
