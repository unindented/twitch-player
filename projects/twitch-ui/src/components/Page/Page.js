import PropTypes from "prop-types";
import React from "react";
import { StyleSheet, View } from "react-native";
import { useLayout } from "../../hooks";
import Main from "../Main";
import Navigation from "../Navigation";

const layoutQuery = {
  narrow: {
    maxWidth: 480,
  },
  wide: {
    minWidth: 481,
  },
};

const Page = ({ children, testID = "page" }) => {
  const [layout, updateLayout] = useLayout(layoutQuery);

  return (
    <View
      style={[styles.root, styles[layout || "narrow"]]}
      testID={testID}
      onLayout={updateLayout}
    >
      {layout && (
        <>
          <Navigation />
          <Main>{children}</Main>
        </>
      )}
    </View>
  );
};

Page.propTypes = {
  children: PropTypes.node.isRequired,
  testID: PropTypes.string,
};

const styles = StyleSheet.create({
  root: {
    flex: 1,
  },
  narrow: {
    flexDirection: "column",
  },
  wide: {
    flexDirection: "row",
  },
});

export default Page;
