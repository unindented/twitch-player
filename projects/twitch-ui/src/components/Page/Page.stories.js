import { storiesOf } from "@storybook/react";
import React from "react";
import Spinner from "../Spinner";
import Page from ".";

storiesOf("Components|Page", module).add("default", () => (
  <Page>
    <Spinner />
  </Page>
));
