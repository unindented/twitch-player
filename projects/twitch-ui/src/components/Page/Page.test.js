import { render, fireResizeEvent } from "@twitch-player/testing/dist/unit";
import React from "react";
import { Text } from "react-native";
import Page from "./Page";

jest.mock("../Navigation", () => "mock-navigation");
jest.mock("../Main", () => "mock-main");

describe("Page", () => {
  let instance;

  beforeEach(() => {
    instance = render(
      <Page>
        <Text>OHAI</Text>
      </Page>
    );
  });

  it("renders nothing until layout", () => {
    expect(instance.getByTestId("page")).toMatchSnapshot();
  });

  describe("with narrow parent", () => {
    beforeEach(() => {
      fireResizeEvent(window, { width: 320, height: 640 });
      jest.runAllTimers();
    });

    it("renders with a narrow layout", () => {
      expect(instance.getByTestId("page")).toMatchSnapshot();
    });
  });

  describe("with wide parent", () => {
    beforeEach(() => {
      fireResizeEvent(window, { width: 640, height: 320 });
      jest.runAllTimers();
    });

    it("renders with a wide layout", () => {
      expect(instance.getByTestId("page")).toMatchSnapshot();
    });
  });
});
