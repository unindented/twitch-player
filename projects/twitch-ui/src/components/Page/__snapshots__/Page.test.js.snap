// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders nothing until layout 1`] = `
<div
  class="css-view-1dbjc4n r-flex-13awgt0"
  data-testid="page"
/>
`;

exports[`Page with narrow parent renders with a narrow layout 1`] = `
<div
  class="css-view-1dbjc4n r-flex-13awgt0 r-flexDirection-eqz5dr"
  data-testid="page"
>
  <mock-navigation />
  <mock-main>
    <div
      class="css-text-76zvg2"
      dir="auto"
    >
      OHAI
    </div>
  </mock-main>
</div>
`;

exports[`Page with wide parent renders with a wide layout 1`] = `
<div
  class="css-view-1dbjc4n r-flex-13awgt0 r-flexDirection-18u37iz"
  data-testid="page"
>
  <mock-navigation />
  <mock-main>
    <div
      class="css-text-76zvg2"
      dir="auto"
    >
      OHAI
    </div>
  </mock-main>
</div>
`;
